homeassistant/components/nasweb/__init__.py

```diff
@@ -0,0 +1,120 @@
+"""The NASweb integration."""
+
+from __future__ import annotations
+
+import logging
+
+from webio_api import WebioAPI
+from webio_api.api_client import AuthError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.network import NoURLAvailableError
+from homeassistant.util.hass_dict import HassKey
+
+from .const import DOMAIN, MANUFACTURER, SUPPORT_EMAIL
+from .coordinator import NASwebCoordinator
+from .nasweb_data import NASwebData
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+NASWEB_CONFIG_URL = "https://{host}/page"
+
+_LOGGER = logging.getLogger(__name__)
+type NASwebConfigEntry = ConfigEntry[NASwebCoordinator]
+DATA_NASWEB: HassKey[NASwebData] = HassKey(DOMAIN)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: NASwebConfigEntry) -> bool:
+    """Set up NASweb from a config entry."""
+
+    if DATA_NASWEB not in hass.data:
+        data = NASwebData()
+        data.initialize(hass)
+        hass.data[DATA_NASWEB] = data
+    nasweb_data: NASwebData = hass.data[DATA_NASWEB]
```

Comment:
When using `HassKey`, there's no need to explicitly tell the type checker the type. Please remove it here and in other places.

```suggestion
    nasweb_data = hass.data[DATA_NASWEB]
```